### ISSUE TITLE
:dependsOn does not load transitive dependencies

### ISSUE THREAD

**Issue by NorbertSandor** [Kotlin/kotlin-interactive-shell#109]
It seems that only the direct dependency is loaded, its transitive dependencies are not.
This makes even slightly complex dependencies unusable without lots of manual work :(

Embedding [Maven Artifact Resolver](https://maven.apache.org/resolver/) would probably help a lot in the implementation.

**Comment by asm0dey** [comment]
Hi @NorbertSandor!

Could you give us an example please? Adding Maven Artefact Resolver is definitely not an option: it's too heavyweight (we've been considering it for some time)

**Comment by NorbertSandor** [comment]
> Adding Maven Artefact Resolver is definitely not an option

Without transitive dependency resolution, `:dependsOn` is useable only for very basic libraries :(

**Comment by asm0dey** [comment]
But they all are available in the repository? May I ask you to give me coordinates of the dependency please?

**Comment by NorbertSandor** [comment]
It was only in my local Maven repository. It seemed that the library itself has been found but its dependencies were not resolved (`ClassNotFoundError` or `NoClassDefFoundError` at runtime when I tried to invoke functions from the library - sorry, I don't remember which one).

**Comment by asm0dey** [comment]
Yeah, maybe it's an issue with it being local repo. I'm almost sure it works for remote ones.
Still an issue tho
